json-endpoints/jetpack/class.jetpack-json-api-themes-install-endpoint.php

```diff
@@ -14,6 +14,12 @@ protected function install() {
 
 		foreach ( $this->themes as $theme ) {
 
+			// hook to allow alternative install method
+			if ( wp_endswith( $theme, '-wpcom' )
+				 && apply_filters( 'jetpack_wpcom_theme_install', false, $theme ) ) {
+				continue;
```

Comment:
Also, if the theme installation failed in `jetpack_wpcom_theme_install` hook, there is no way to return a `WP_Error`. Normally, it's returned on line [35](https://github.com/Automattic/jetpack/blob/7cf4e71461ef2b5ef2c67eb028f5d3810bf01fca/json-endpoints/jetpack/class.jetpack-json-api-themes-install-endpoint.php#L35).